content/blog/jetstream-java-client-03-consume.md

```diff
@@ -0,0 +1,131 @@
+
+### Deliver Subject
+
+Deliver subject essentially creates an alias core NATS subject for the stream. This means a core NATS subscriber could be set up to receive messages on the deliver subject,
+starting wherever the deliver policy was configured for the consumer. You could use a core NATS subscriber to access the stream by its original subject, but that would always start at
+the next message that appears on the subject, instead of where the consumer configured start sequence or start time. This is only allowed for push subscriptions.
+
+### Ack Policy
+
+`Explicit` is the default policy and means that each individual message must be acknowledged.
+For pull consumers, `Explicit` is the only allowed option.
+For push consumers, you can choose `None`, which means you do not have to ack any messages,
+or `All` which means whenever you choose to ack a message, all the previous messages received are automatically acknowledged.
+You must ack within the Ack Wait window or it will be as if the ack was not received.
+
+### Ack Wait
+
+Ack Wait is the time in nanoseconds that the server will wait for an ack for any individual message.
+If an ack is not received in time, the message will be redelivered.
+
+### Max Ack Pending
+
+If you have received this number of messages without acking them, the server will stop sending messages.
+If the server gets 1 or more acks, it will resume sending that number of messages.
+If acks don't occur in the Ack Wait period, then the server will resume starting at the messages that have not been acked.
+
+### Replay Policy
+
+The replay policy applies when the deliver policy is `All`, `ByStartSequence` or `ByStartTime` since those deliver policies begin reading the stream at a position other than the end.
+If the policy is `ReplayOriginal` (the default), the messages in the stream will be pushed to the client at the same rate that they were originally received, simulating the original timing of messages.
+If the policy is `ReplayInstant`, the messages will be pushed to the client as fast as possible while adhering to the Ack Policy, Max Ack Pending and the client's ability to consume those messages.
+
+### Max Deliver
+
+The maximum amount times a specific message will be delivered. Applies to messages that are re-sent due to ack policy.
+
+### Filter Subject
+
+When consuming from a stream with many subjects, or wildcards, this allows you to select only a specific incoming subjects, supports wildcards
+
+### Rate Limit
+
+Used to throttle the delivery of messages to the consumer, in bits per second.
+
+### Sample Frequency
+
+Sets the percentage of acknowledgements that should be sampled for observability, 0-100
+
+### Idle Heartbeat
+
+If the idle heartbeat period is set, the server will send a status message to the client when the period has elapsed but it has not received any new messages.
+This lets the client know that it's still there, but just isn't receiving messages.
```

Comment:
Is the API handling it internally on behalf of the user? If not we have to include here what the code is (100) and that its a empty message

Reply:
no the client is not. Will add clarification